docs/proposals/cloud-edge-custom-message-design.md

```diff
@@ -0,0 +1,429 @@
+                    when ruleEndpointType is servicebus, its value is {"service_port":"8080"}.'
+                  type: object
+                  additionalProperties:
+                    type: string
+              required:
+                - ruleEndpointType
+  scope: Namespaced
+  names:
+    plural: ruleendpoints
+    singular: ruleendpoint
+    kind: RuleEndpoint
+    shortNames:
+      - re
+
+```
+
+* Rule 
+```
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: rules.rules.kubeedge.io
+spec:
+  group: rules.kubeedge.io
+  versions:
+    - name: v1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                source:
+                  description: 'source is a string value representing where the messages come from. Its
+                    value is the same with rule-endpoint name. For example, my-rest or my-eventbus.'
+                  type: string
+                sourceResource:
+                  description: 'sourceResource is a map representing the resource info of source. For rest
+                    rule-endpoint type its value is {"path":"/a/b"}. For eventbus rule-endpoint type its
+                    value is {"topic":"<user define string>","node_name":"xxxx"}'
+                  type: object
+                  additionalProperties:
+                    type: string
+                target:
+                  description: 'target is a string value representing where the messages go to. its value is
+                     the same with rule-endpoint name. For example, my-eventbus or my-rest or my-servicebus.'
+                  type: string
+                targetResource:
+                  description: 'targetResource is a map representing the resource info of target. For rest
+                    rule-endpoint type its value is {"resource":"http://a.com"}. For eventbus rule-endpoint
+                    type its value is {"topic":"/xxxx"}. For servicebus rule-endpoint type its value is {"path":"/request_path"}.'
+                  type: object
+                  additionalProperties:
+                    type: string
+              required:
+                - source
+                - sourceResource
+                - target
+                - targetResource
+            status:
+              type: object
+              properties:
+                successMessages:
+                  type: integer
+                failMessages:
+                  type: integer
+                errors:
+                  items:
+                    type: string
+                  type: array
+              required:
+                - successMessages
+                - failMessages
+                - errors
+  scope: Namespaced
+  names:
+    plural: rules
+    singular: rule
+    kind: Rule
+    shortNames:
+      - rule
```

Comment:
here rule shortnames not needed, because they are the same.

Reply:
ok